### PR TITLE
Make SIM119 not trigger on complex operations in the constructor, and when async methods are present on a class.

### PR DESCRIPTION
**Pull request by dsegan**
Closes issue #63.

### CODE REVIEW

File: flake8_simplify.py [hunk 1]
```diff
@@ -1093,24 +1093,39 @@ def _get_sim119(node: ast.ClassDef) -> List[Tuple[int, int, str]]:
...
-        if (
-            isinstance(body_el, ast.FunctionDef)
-            and body_el.name not in dataclass_functions
-        ):
-            has_only_constructur_function = False
-            break
+        if isinstance(body_el, (ast.FunctionDef, ast.AsyncFunctionDef)):
+            has_any_functions = True
+            if body_el.name == "__init__":
+                # Ensure constructor only has pure assignments without any calculation.
```
Comment (MartinThoma) [inline flake8_simplify.py:1102]:
```suggestion
                # Ensure constructor only has pure assignments
                # without any calculation.
```

File: flake8_simplify.py [hunk 2]
```diff
@@ -1093,24 +1093,39 @@ def _get_sim119(node: ast.ClassDef) -> List[Tuple[int, int, str]]:
...
+                has_complex_statements = False
+                for el in body_el.body:
+                    if not isinstance(el, ast.Assign):
+                        has_complex_statements = True
+                        break
+                    else:
+                        # It is an assignment, but we only allow `self.attribute = name`.
```
Comment (MartinThoma) [inline flake8_simplify.py:1109]:
```suggestion
                        # It is an assignment, but we only allow
                        # `self.attribute = name`.
```

File: tests/test_simplify.py [hunk 3]
```diff
@@ -476,6 +476,19 @@ def test_sim118_del_key():
...
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+"""
+    )
+    assert results == {"2:0 SIM119 Use a dataclass for 'class FooBar'"}
+
+
+def test_sim119_ignored_dunder_methods():
+    """Dunder methods like __str__ do not make a class not be a dataclass candidate."""
```
Comment (MartinThoma) [inline tests/test_simplify.py:489]:
```suggestion
    """
    Dunder methods do not make a class not be a dataclass candidate.
    
    Examples for dunder (double underscore) methods are:
      * __str__
      * __eq__
      * __hash__
    """
```

**Comment by MartinThoma** [comment]
@dsegan Thank you very much for the PR! Looks good - I just have some minor changes to keep the maximum line length.

File: tests/test_simplify.py [hunk 4]
```diff
@@ -476,6 +476,26 @@ def test_sim118_del_key():
...
+        self.b = b
+"""
+    )
+    assert results == {"2:0 SIM119 Use a dataclass for 'class FooBar'"}
+
+
+def test_sim119_ignored_dunder_methods():
+    """
+    Dunder methods do not make a class not be a dataclass candidate.
+
```
Comment (MartinThoma) [inline tests/test_simplify.py:491]:
```suggestion

```

**Comment by dsegan** [comment]
@MartinThoma Thanks a bunch for the quick review, and my apologies for the line length omissions!

**Pull request merged**